mlx_graphs/datasets/ogb_dataset.py

```diff
@@ -0,0 +1,204 @@
+    Node property prediction datasets consist of a single graph with three additional
+    properties: `train_mask`, `val_mask` and `test_mask` specifying the masks for the
+    train, validation and test splits.
+    Link property prediction datasets also consist of a single graphs with three
+    additional properties: `train_edge_index`, `val_edge_index` and `test_edge_index`,
+    specifying the edges to be considered for training, validation and testing.
+    Graph property prediction datasets consists of multiple graphs. The desired split
+    can be specified via the `split` arg.
+
+    See `here <https://ogb.stanford.edu/docs/dataset_overview/>`_ for further details
+    and a list of the available datasets with their descriptions
+
+    Args:
+        name: Name of the dataset
+        split: Split of the dataset to load. Thi parameter has effect only on graph
+            property prediction dataset. If `None`, the entire dataset is loaded.
+            Defaults to `None`.
+        base_dir: Directory where to store dataset files. Default is
+            in the local directory ``.mlx_graphs_data/``.
+
+    .. note::
+
+       `ogb` needs to be installed to use this dataset
+
+    .. note::
+
+        The ogbn-mag, ogbl-wikikg2 and igbl-biokg and the graphs belonging to the
+        largs-scale challenge category are currently not available as
+        they require heterogenous graphs which are not yet supported by `mlx-graphs`
+
+    Example:
+
+    .. code-block:: python
+
+        from mlx_graphs.datasets.ogb_dataset import OGBDataset
+
+        ds = OGBDataset("ogbg-molhiv", split="train")
+        >>> ogbg-molhiv(num_graphs=32901)
+
+    """
+
+    def __init__(
+        self,
+        name: ALL_DATASETS,
+        split: Optional[Literal["train", "val", "test"]] = None,
+        base_dir: Optional[str] = None,
+    ):
+        self.split = split
+        super().__init__(name, base_dir)
+
+    def download(self):
+        self._load_data()
+
+    def _load_data(self):
+        try:
+            if self.name in get_args(OGB_NODE_DATASET):
+                from ogb.nodeproppred import NodePropPredDataset as OGB_dataset
+            elif self.name in get_args(OGB_EDGE_DATASET):
+                from ogb.linkproppred import LinkPropPredDataset as OGB_dataset
+            elif self.name in get_args(OGB_GRAPH_DATASET):
+                from ogb.graphproppred import GraphPropPredDataset as OGB_dataset
+            self._raw_ogb_dataset = OGB_dataset(name=self.name, root=self.raw_path)
+        except ImportError:
+            raise ImportError(
+                "ogb needs to be installed to use this dataset",
+                "you can install it via pip install ogb",
+            )
+
+    def process(self):
+        try:
+            dataset = self._raw_ogb_dataset
+        except AttributeError:
+            # reload if already downloaded
+            self._load_data()
+            dataset = self._raw_ogb_dataset
```

Comment:
isn't it cleaner to declare `self._raw_ogb_dataset = None` in the `__init__` and check if `self._raw_ogb_dataset is None` rather than a `try/except`?

Reply:
I didn't do that to avoid having typing errors, but yes, it's cleaner so I fixed it.